extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/QuarkusResteasyReactiveRequestContext.java

```diff
@@ -113,41 +144,97 @@ private <E extends Throwable> RuntimeException sneakyThrow(Throwable e) throws E
     @Override
     protected void invokeHandler(int pos) throws Exception {
         var handler = handlers[pos];
+        var handlerType = knownHandlerTypeMap[pos];
+        switch (handlerType) {
```

Comment:
The reason is the same behind the original implementation, having separate call-sites (ie calls to `handler.handle(this)`) which, each one, at runtime, can observe a single "received" type (eg one of the types in the case), it could:
- inline `handler.handle(this)`
- no need to perform a megamorphic call there (which could end up using an itable and a vtable to find the appropriate version per implementors)

This PR enable to not performing this lookup, if not once, on handlers initial build.